File: CoWorkHub/UI/coworkhub_mobile/lib/models/notification.dart
import 'package:json_annotation/json_annotation.dart';

part 'notification.g.dart';

@JsonSerializable()
class Notification {
  int notificationId;
  int userId;
  String message;
  bool isRead;
  DateTime createdAt;

  Notification({
    required this.notificationId,
    required this.userId,
    required this.message,
    required this.isRead,
    required this.createdAt,
  });

  factory Notification.fromJson(Map<String, dynamic> json) =>
      _$NotificationFromJson(json);

  Map<String, dynamic> toJson() => _$NotificationToJson(this);
}

File: CoWorkHub/CoWorkHub.Model/Notification.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CoWorkHub.Model
{
    public class Notification
    {
        public int NotificationId { get; set; }
        public int UserId { get; set; }
        public string Message { get; set; } = null!;
        public bool IsRead { get; set; } = false;
        public DateTime CreatedAt { get; set; }
        public virtual User User { get; set; } = null!;
    }
}
